eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/SynchronizedMutableBiMap.java

```diff
@@ -0,0 +1,478 @@
+
+    public CharIterable collectChar(CharFunction<? super V> charFunction)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectChar(charFunction);
+        }
+    }
+
+    public DoubleIterable collectDouble(DoubleFunction<? super V> doubleFunction)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectDouble(doubleFunction);
+        }
+    }
+
+    public FloatIterable collectFloat(FloatFunction<? super V> floatFunction)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectFloat(floatFunction);
+        }
+    }
+
+    public IntIterable collectInt(IntFunction<? super V> intFunction)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectInt(intFunction);
+        }
+    }
+
+    public LongIterable collectLong(LongFunction<? super V> longFunction)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectLong(longFunction);
+        }
+    }
+
+    public ShortIterable collectShort(ShortFunction<? super V> shortFunction)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectShort(shortFunction);
+        }
+    }
+
+    public <P, V1> RichIterable<V1> collectWith(Function2<? super V, ? super P, ? extends V1> function, P parameter)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectWith(function, parameter);
+        }
+    }
+
+    public <V1> RichIterable<V1> collectIf(Predicate<? super V> predicate, Function<? super V, ? extends V1> function)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectIf(predicate, function);
+        }
+    }
+
+    public <V1> RichIterable<V1> flatCollect(Function<? super V, ? extends Iterable<V1>> function)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().flatCollect(function);
+        }
+    }
+
+    public <V1> MutableSetMultimap<V1, V> groupBy(Function<? super V, ? extends V1> function)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().groupBy(function);
+        }
+    }
+
+    public <V1> MutableSetMultimap<V1, V> groupByEach(Function<? super V, ? extends Iterable<V1>> function)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().groupByEach(function);
+        }
+    }
+
+    public MutableSetMultimap<V, K> flip()
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().flip();
+        }
+    }
+
+    public MutableBiMap<K, V> newEmpty()
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().newEmpty();
+        }
+    }
+
+    public MutableBiMap<V, K> inverse()
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().inverse();
```

Comment:
This can't be safe. You can mutate this.delegate by just calling inverse() and mutating the inverse, no?